lib/api/helpers/namespaces.rb

```diff
@@ -8,6 +8,10 @@ def can_manage_namespace?(namespace, user)
         NamespacePolicy.new(user, namespace).update?
       end
 
+      def can_destroy_namespace?(namespace, user)
+        NamespacePolicy.new(user, namespace).destroy?
```

Comment:
@vitoravelino all that would take is to update this method so it also checked personal namespaces, no?